Packs/GuardiCore/Integrations/GuardiCoreV2/GuardiCoreV2.py

```diff
@@ -0,0 +1,448 @@
+
+        self.login()
+
+    def login(self):
+        integration_context = get_integration_context()
+
+        if self.is_access_token_valid(integration_context):
+            access_token = integration_context.get('access_token')
+            self.save_access_token(access_token)
+        else:
+            demisto.debug(
+                "GuardicoreV2 - Generating a new token (old one isn't valid anymore).")
+            self.generate_new_token()
+
+    def save_access_token(self, access_token: str):
+        self.access_token = access_token
+        authorization_value = f'bearer {access_token}'
+        self._headers = {
+            "Authorization": authorization_value}
+
+    def is_access_token_valid(self, integration_context: dict):
+        access_token_expiration = integration_context.get('expires_in')
+        access_token = integration_context.get('access_token')
+        demisto.debug(
+            'GuardicoreV2 - Checking if context has valid access token...'
+            + f'expiration: {access_token_expiration}, access_token: {access_token}')
+        if access_token and access_token_expiration:
+            access_token_expiration_datetime = datetime.strptime(
+                access_token_expiration, DATE_FORMAT)
+            return access_token_expiration_datetime > datetime.now()
+        return False
+
+    def generate_new_token(self):
+        token = self.authenticate()
+        self.save_jwt_token(token)
+        self.save_access_token(token)
+
+    def save_jwt_token(self, access_token: str):
+        expiration = get_jwt_expiration(access_token)
+        expiration_timestamp = datetime.fromtimestamp(expiration)
+        context = {"access_token": access_token,
+                   "expires_in": expiration_timestamp.strftime(DATE_FORMAT)}
+        set_integration_context(context)
+        demisto.debug(
+            f"New access token that expires in : {expiration_timestamp.strftime(DATE_FORMAT)}"
+            f" was set to integration_context.")
+
+    def authenticate(self):
+        body = {
+            'username': self.username,
+            'password': self.password
+        }
+        new_token = self._http_request(
+            method='POST',
+            url_suffix='/authenticate',
+            json_data=body)
+
+        if not new_token or not new_token.get('access_token'):
+            raise DemistoException(
+                "GuardiCore error: The client credentials are invalid.")
+
+        new_token = new_token.get('access_token')
+        return new_token
+
+    def get_assets(self, url_params: dict):
+        data = self._http_request(
+            method='GET',
+            url_suffix='/assets',
+            params=url_params
+        )
+        return data
+
+    def get_incident(self, url_params: str):
+        data = self._http_request(
+            method='GET',
+            url_suffix=f'/incidents/{url_params}',
+        )
+        return data
+
+    def get_incidents(self, url_params: dict):
```

Comment:
how many incidents can there be? can we reach a timeout? the default of _http_request is 10seconds. consider adding an argument for that.